### DISCUSSION TITLE
Trouble with generators and iterators

### DISCUSSION THREAD

**Discussion by brainkim** [codesandbox/codesandbox-client#4074] [General]
I’m consistently having strange bugs related to Crank.js projects in codesandbox. The projects will work in a parcel 1 app locally, but when I try to upload them to codesandbox they stop working.

For instance take this sandbox: https://codesandbox.io/s/crankjs-threejs-boxes-demo-ypyqi?file=/src/crank-three.tsx:5867-5904

Works locally, but once I copied the files into the sandbox, iteration over this just stopped working for one file. Weird I think, maybe they’re not downleveling iteration with TypeScript. So I do that and all of a sudden some variable in another scope is undefined, when it’s never unassigned within the loop.

These bugs are scary and completely unintrospectable, and I suspect they have to do with outdated versions of regeneratorRuntime/babel. Is there somewhere I can get more information/debug what the code is actually being compiled to?

**Comment by CompuIves** [comment]
Hmm, this is indeed very weird. It could maybe be because we use the `main` module instead of the esmodule in Three context. I've seen discussions around this before. Is this only in Three projects, or any project?

**Reply by brainkim** [reply]
I’ve also seen a similar issue where I tried to destructure an iterator in a variable `for ({variable} of this)` where, if the variable name was different it wouldn’t error, as in, it would only error for a specific identifier. I don’t remember the variable name, and I apologize for not capturing and shrinking the bug when I found it. I suspect these are issues related to code sandbox’s compilation pipeline? How would I begin investigating this in the repository? If I encounter any more bugs I will definitely try to shrink them, I just didn’t have the time.
